app-office/texmacs/texmacs-1.99.12.recipe

```diff
@@ -0,0 +1,102 @@
+SUMMARY="A scientific editing platform designed to create beautiful technical documents"
+DESCRIPTION="TeXmacs provides a unified and user friendly framework for editing \
+structured documents with different types of content: text, mathematics, graphics, \
+interactive content, slides, etc.\
+Its rendering engine uses high-quality typesetting algorithms so as to produce \
+professionally looking documents, which can either be printed out or presented from \
+a laptop."
+HOMEPAGE="http://www.texmacs.org/"
+COPYRIGHT="Free Software Foundation"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/mgubi/texmacs/archive/texmacs-$portVersion-haiku-qt5.tar.gz"
+CHECKSUM_SHA256="607fa4db4676b78780f7af086feec2e2332dc18a371d79fa50d640de6db10b43"
+SOURCE_DIR="texmacs-texmacs-$portVersion-haiku-qt5/src"
+PATCHES="texmacs-$portVersion.patchset"
+ADDITIONAL_FILES="texmacs.rdef.in"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	texmacs$secondaryArchSuffix = $portVersion
+	app:TeXmacs
+	"
+
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libQt5Concurrent$secondaryArchSuffix
+	lib:libQt5Core$secondaryArchSuffix
+	lib:libQt5Gui$secondaryArchSuffix
+	lib:libQt5Svg$secondaryArchSuffix
+	lib:libQt5Widgets$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	lib:libguile$secondaryArchSuffix <= 19
+	lib:libfreetype$secondaryArchSuffix
+	lib:libgmp$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
+	lib:libltdl$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libiconv$secondaryArchSuffix
+	devel:libQt5Concurrent$secondaryArchSuffix
+	devel:libQt5Core$secondaryArchSuffix
+	devel:libQt5Gui$secondaryArchSuffix
+	devel:libQt5Svg$secondaryArchSuffix
+	devel:libQt5Widgets$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	devel:libguile$secondaryArchSuffix <= 19
```

Comment:
Should be 17

Reply:
(see below) Guile 2.0 has libraries with version 20. This is the first version which does not work with TeXmacs, all the lower versioned libraries are ok.

Reply:
Shouldn't be needed, Guile 1.8 provides "lib:libguile$secondaryArchSuffix", Guile 2.2.4 provides "lib:libguile_2.2$secondaryArchSuffix" so only using lib/devel:libguile$secondaryArchSuffix should be enough ...

Reply:
It will not harm anyway and it would protect from some other package providing libguile with higher version. I would tend to keep. There is no advantage in removing it.

Reply:
Then I tend to agree with @extrowerk here, keep the version provided by Guile 1.8 (17), Guile doesn't provide libguile (but libguile-2.2)

Reply:
Ok. I will change.